### PR TITLE
chore/add-typescript

### LINKED ISSUE #120: Update JavaScript Files to Typescript 🎯
Purpose: Update all files to typescript, and type all properties.

### PR DESCRIPTION
**Pull request by ramonxm**

### CODE REVIEW

**Comment by ramonxm** [comment]
Draft..

**Comment by Brenont** [comment]
> Draft..

@ramonxm you can open PR as a draft when opening or edit

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

**Comment by ramonxm** [comment]
Thankss!✌🏻✌🏻

File: .eslintrc.json [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
+    ],
+    "rules": {
+      "import/prefer-default-export": "off",
+      "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx", ".ts", ".tsx"] }],
+      "react/react-in-jsx-scope": "off",
+      "react/jsx-props-no-spreading": "off",
+      "react/prop-types": "off",
+      "react/require-default-props": "off",
+      "no-bitwise": "off",
+      // "jest-dom/prefer-checked": "error",
```
Comment (jonataspinto) [inline .eslintrc.json:32]: comentarios

**Review by joaoprferreira — APPROVED** [review]
@ramonxm da próxima vez que for abrir um mr quebra ele em pequenas partes, fora isso estar indo bem demais. 🔥 🚀

**Pull request merged**